### ISSUE TITLE
Content development draft version and preview

### ISSUE THREAD

**Issue by lucyWMDE** [codingdavinci/relaunch2018#105]
I can't figure out how to see a preview version of a page under development. Once the site is live we need to be able to work on and agree draft versions of pages before finally publishing them. I assume this should work by saving as Entwurf... this works to save content changes in the GUI, but I don't see the Entwurf version in the logged-in front-end preview, just the latest published status.
Since the WYSIWYG editor version doesn't mirror exactly the front-end we really need this preview if we don't want to end up developing in the live site.

**Comment by Snater** [comment]
I'm sorry, I am not sure. All content is listed on the "Content" page. You can click the title and will see the latest revision with controls to alter the content moderation state.
On that "preview" view—which should be the latest revision, regardless of whether it is published or still a draft—, you can also access all previous revisions clicking the corresponding link on top of the page, which is where I have noticed an issue about those JavaScript menus—they may be on top of that link (will look into that). That list of revisions may also be accessed per the "Revisionen" tab on the edit view of a node.

**Comment by lucyWMDE** [comment]
ok. Then perhaps I don't know how to save a draft. Here's a use-case:
I'm using the admin account... I open the Über page "Was ist Coding da Vinci?" in Bearbeitungsmodus and add an additional paragraph at the bottom. 
- If I directly click the "Ansicht tab" I don't see any change and I loose my edits. 
- If I click save it publishes the changes to the live page
- If I deselect the "published" häckchen then save, the page no longer exists on the live site.

I don't see an option to save a draft version whilst leaving the last published version on the live site.

**Comment by lucyWMDE** [comment]
Could be that this is only a problem for Einzelne Seiten... 
Checking other page types...
- [ ] dataset
- [ ] project
- [ ] event
- [ ] FAQ 
- [ ] pressemappe
- [ ] berichterstattung
- [ ] Artikel

**Comment by lucyWMDE** [comment]
So. 
- Einfache Seiten definitely need the possibility of a draft version. 
- Berichterstattung doesn't have it, but that's fine. 
- If it's easy it would be nice to have it for Pressemappe as there's some minor styling you need to do there. If it's not easy, then no worries - I bet we mainly copy-past the same style over and over.
- FAQ doesn't have it but could also use it if it's easy to implement. I could imagine we might rework that page at some point and it'd be cool to have a draft version for freigabe in the correct layout.

**Comment by Snater** [comment]
OK, now I get the problem, I guess I was just confused by the term page (which was technically correct though, because that's the English/machine name of the "Einfach Seite" content type).
I have added content moderation to all other content types except Berichterstattung. I actually wondered why some content types except Event, data set and project hat content moderation activated. Might actually have been injected by some Drupal core update (as far as I remember we had invoked the content moderation module before it was finally released) as the Geschäftsstelle role was actually not enabled to apply those content moderation state changes…
I hope it's working as expected now. Just reopen, if not.

**Comment by lucyWMDE** [comment]
ok. testing again...

- [ ] dataset
- [ ] project
- [ ] event
- [ ] FAQ 
- [ ] pressemappe
- [ ] berichterstattung
- [ ] Artikel 
- [ ] Einfache Seite

**Comment by lucyWMDE** [comment]
No possibility for admin to save as Entwurf - I guess this is fine:
- FAQ
- Pressemappe
- Berichterstattung

Works for admin to make edits, save as Entwurf, then publish:
Dataset, Projekt, Event, Einfache Seite, Artikel

... but:
For an Event page created by testmanager2, editing and saving as Entwurf as admin, then publishing in a second step, doesn't work. In this case, admin saving option defaulty to save as Entwurf, but the Entwurf is actually Veröffentlicht... will have to check through all content types with the relevant users.

**Comment by Snater** [comment]
For FAQ, Pressemappe, Berichterstattung, the moderation state form element is deactivated in the form, so that behaviour is expected, though it could easily be enabled.
I am not sure I get the "but" issue, I don't notice faulty behaviour. Those two "adfgadf" events currently in the content are not published. I saved one as Veröffentlicht and it got published. Then, when editing again, I have the choice to keep Veröffentlicht, submit another Entwurf (keeping the last Veröffentlicht version public), or switch to Unveröffentlicht which would completely unpublish the node, so it would not be accessible.

**Comment by lucyWMDE** [comment]
basically happy with the conclusion here... though I'd like to be able to make draft version AND unpublish FAQs. I figured out how to add the "moderation state" element under struktur/inhaltstypen/FAQ/Formularanzeige Verwalten and now have the option to save as Entwurf. However, I can't figure out how to get the option to unpublish. 
- I added element "Published" but that didn't seem to do it.
- I unchecked the box "Veröffentlicht" under struktur/inhaltstypen/FAQ/bearbeiten/Veröffentlichungseinstellungen but that didn't do it either...

pls halp
